dd-java-agent/instrumentation/java-lang/src/main/java/datadog/trace/instrumentation/java/lang/StringCallSite.java

```diff
@@ -281,4 +282,74 @@ public static String[] afterSplitWithLimit(
     }
     return result;
   }
+
+  @CallSite.After("java.lang.String java.lang.String.replace(char, char)")
+  public static String afterReplaceChar(
+      @CallSite.This @Nonnull final String self,
+      @CallSite.Argument(0) final char oldChar,
+      @CallSite.Argument(1) final char newChar,
+      @CallSite.Return @Nonnull final String result) {
+    final StringModule module = InstrumentationBridge.STRING;
+    if (module != null) {
+      try {
+        module.onStringReplace(self, oldChar, newChar, result);
+      } catch (final Throwable e) {
+        module.onUnexpectedException("afterReplaceChar threw", e);
+      }
+    }
+    return result;
+  }
+
+  @CallSite.Around(
+      "java.lang.String java.lang.String.replace(java.lang.CharSequence, java.lang.CharSequence)")
+  public static String aroundReplaceCharSeq(
+      @CallSite.This final String self,
+      @CallSite.Argument(0) final CharSequence oldCharSeq,
+      @CallSite.Argument(1) final CharSequence newCharSeq) {
+    final StringModule module = InstrumentationBridge.STRING;
+    if (module != null) {
+      try {
+        return module.onStringReplace(self, oldCharSeq, newCharSeq);
+      } catch (final Throwable e) {
+        module.onUnexpectedException("aroundReplaceCharSeq threw", e);
+      }
+    }
+    return self.replace(oldCharSeq, newCharSeq);
```

Comment:
We don't want to be in the stack if the original function fails:

```suggestion
    try {
        return self.replace(oldCharSeq, newCharSeq);
    } catch (Throwable e) {
        throw StackUtils.filterDatadog(e);
    }     
```

Reply:
Done!